pkg/dataplane/client.go

```diff
@@ -1,19 +1,59 @@
 package dataplane
 
 import (
+	"context"
+	"fmt"
+	"reflect"
+	"regexp"
+
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
 	"github.com/Azure/msi-dataplane/internal/swagger"
+	"github.com/go-playground/validator/v10"
 )
 
-// TODO - Tie the module version to update automatically with new releases
-const moduleVersion = "v0.0.1"
+//go:generate /bin/bash -c "../../hack/mockgen.sh mock_swagger_client/zz_generated_mocks.go client.go"
+
+const (
+	// TODO - Tie the module version to update automatically with new releases
+	moduleVersion = "v0.0.1"
+
+	// This regular expression is used to validate the resource ID
+	resourceIDRegex = `^/subscriptions/[a-z0-9\-]+/resourceGroups/[a-z0-9\-]+/providers/Microsoft\.ManagedIdentity/userAssignedIdentities/[A-Za-z0-9\-]+$`
```

Comment:
I should have mentioned this earlier, but we could also use `github.com/Azure/azure-sdk-for-go/sdk/azcore/arm` to validate the regex here: https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azcore/arm#ParseResourceID 
```
_, err := azcorearm.ParseResourceID(resourceIDRegexp)
if err != nil {
     return err
}
```

Reply:
Thank you! Didn't know about this.

Reply:
I vote to keep the regex because it's more specific to the resource ID format for UA MSI. For example, let's say we pass as a resource ID for another resource.

Reply:
Oh wait just noticed that `arm` package has checks for resource type as well! Will adopt that.